src/plugins/api/BioSignalMLAPI/CMakeLists.txt

```diff
@@ -1,77 +1,219 @@
+            RETRIEVE_PACKAGE_FILE(${PACKAGE_NAME} ${PACKAGE_VERSION}
+                                  ${RELATIVE_PROJECT_SOURCE_DIR} e2ca2a020750760c611b72b35edfa956bc55c593
+                                  PACKAGE_REPOSITORY ${PACKAGE_REPOSITORY}
+                                  RELEASE_TAG ${RELEASE_TAG}
+                                  SHA1_FILES ${SHA1_FILES}
+                                  SHA1_VALUES 379e38afc9eb015b54cd327aac0d725f25fb95fc
+                                              c18a129f49429083c40c3621f61256ddcbe83c05
+                                              c4bf89a74703d73220e7b8be29944c409fca1a06
+                                              a10d81c681fae3efc41d7855c4250f7333821256)
+        ELSE()
+            RETRIEVE_PACKAGE_FILE(${PACKAGE_NAME} ${PACKAGE_VERSION}
+                                  ${RELATIVE_PROJECT_SOURCE_DIR} f01b5c1219ab792476ae6eaadf3be0329784c948
+                                  PACKAGE_REPOSITORY ${PACKAGE_REPOSITORY}
+                                  RELEASE_TAG ${RELEASE_TAG}
+                                  SHA1_FILES ${SHA1_FILES}
+                                  SHA1_VALUES 6ffeea24b14d9769e5b50aed663ce5091cd43101
+                                              136f63617ad96a272d56b0e13f3d0b090f096d75
+                                              0dcde52f7037472d6d642e71f9e616f0fac18095
+                                              556aa50cf56cd0e934a5cd7c392cef4be1d01d1c)
+        ENDIF()
+    ELSEIF(APPLE)
+        RETRIEVE_PACKAGE_FILE(${PACKAGE_NAME} ${PACKAGE_VERSION}
+                              ${RELATIVE_PROJECT_SOURCE_DIR} a9842f187002c5b9bae273bc2cd3224592f87ef2
+                              PACKAGE_REPOSITORY ${PACKAGE_REPOSITORY}
+                              RELEASE_TAG ${RELEASE_TAG}
+                              SHA1_FILES ${SHA1_FILES}
+                              SHA1_VALUES 7141316b23d7cace02e72466808f6710ad40a4b8
+                                          8e6ca96135ae670a53422a44827579740108d559)
+    ELSE()
+        RETRIEVE_PACKAGE_FILE(${PACKAGE_NAME} ${PACKAGE_VERSION}
+                              ${RELATIVE_PROJECT_SOURCE_DIR} 02a4ba32fce20cd13f9f5fbf966000ce0abb2b9b
+                              PACKAGE_REPOSITORY ${PACKAGE_REPOSITORY}
+                              RELEASE_TAG ${RELEASE_TAG}
+                              SHA1_FILES ${SHA1_FILES}
+                              SHA1_VALUES ecb1f7171e724140b57114c5de6af8b0f36871de
+                                          6d80493dc3187a71ac3755ef689191d50325dce4)
+    ENDIF()
 ELSE()
-    SET(EXTERNAL_BINARIES_DEPENDENCIES)
+
+    SET(PACKAGE_BUILD ${PACKAGE_NAME}Build)
+
+    # First build typedobject library as an external project
+
+    ExternalProject_Add(TypedObjectBuild
+        SOURCE_DIR
+            ${CMAKE_SOURCE_DIR}/ext/typedobject
+        INSTALL_DIR
+            ${FULL_LOCAL_EXTERNAL_PACKAGE_DIR}
+        GIT_REPOSITORY
+            git@github.com:opencor/typedobject.git
```

Comment:
For consistency with our other packages:

`https://github.com/opencor/typedobject`